sources/packages/backend/apps/queue-consumers/src/services/student-assessment/student-assessment.service.ts

```diff
@@ -55,4 +60,42 @@ export class StudentAssessmentService {
       workflowName: data.workflowName,
     };
   }
+
+  /**
+   * Retrieve assessment cancellations to be retried up to a date.
+   * @param retryMaxDate max date to retrieve assessment cancellations to be retried.
+   * @returns student assessment cancellations to be retried.
+   */
+  async getAssessmentCancellationsRequestedToBeRetried(
+    retryMaxDate: Date,
+  ): Promise<StudentAssessment[]> {
+    return this.studentAssessmentRepo.find({
+      select: {
+        id: true,
+      },
+      where: {
+        studentAssessmentStatus: StudentAssessmentStatus.CancellationQueued,
+        updatedAt: LessThan(retryMaxDate),
```

Comment:
The check must happen using `student_assessment_status_updated_on` as per defined in the ticket.